app/main.py

```diff
@@ -1,2 +1,224 @@
+
+    def __eq__(self, other: Dictionary) -> bool:
+        """
+        Compare key, value pairs between two Dictionaries.
+        """
+        if not isinstance(other, Dictionary):
+            return False
+
+        if self.size != other.size:
+            return False
+
+        for node in self.hash_table:
+            if node is not None:
+                if other.__getitem__(node.key) != node.value:
+                    return False
+
+        return True
+
+    def get(self, key: Hashable, default: Optional[Any] = None) -> Any:
+        """
+        Get the value for the given key.
+        If the key does not exist, return the default value.
+
+        Default value is optional.
+        If it was not provided, method will raise KeyError.
+        """
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
+    def pop(self, key: Hashable, default: Optional[Any] = object()) -> Any:
+        try:
+            value = self[key]
+            self.__delitem__(key)
+            return value
+        except KeyError:
+            if isinstance(default, object):
+                return default
+            else:
+                raise KeyError(
+                    f"Cannot delete key: {key} and "
+                    "no default value is provided"
+                )
+
+    def _update_from_dict(self, data: Dictionary) -> None:
+        """
+        Updates the dictionary with a Dictionary object.
+        """
+        for node in data.hash_table:
+            if node is not None:
+                self.__setitem__(node.key, node.value)
+
+    def _update_from_builtin_dict(self, data: dict) -> None:
+        """
+        Updates the dictionary with key-value pairs from builtin dictionary.
+        """
+        for key, value in data.items():
+            self.__setitem__(key, value)
+
+    def update(
+            self,
+            data: Dictionary | list[dict] | dict
+    ) -> None:
+        """
+        Updates Dictionary with various types of data.
+        """
+        if isinstance(data, Dictionary):
+            self._update_from_dict(data)
+        elif isinstance(data, dict):
+            self._update_from_builtin_dict(data)
+        elif isinstance(data, list):
+            for item in data:
+                if isinstance(item, dict):
+                    self._update_from_builtin_dict(item)
+                else:
+                    raise TypeError(f"Unsupported type {type(item)}")
+        else:
+            raise TypeError(f"Unsupported type {type(data)}")
+
+    def copy(self) -> Dictionary:
+        """
+        Creates a copy of the dictionary.
+        """
+        new_dict = Dictionary()
+
+        for node in self.hash_table:
+            if node is not None:
+                new_dict.__setitem__(node.key, node.value)
+
+        return new_dict
+
+    def clear(self) -> None:
+        """
+        Resets the dictionary to its initial state.
+        """
+        self.__init__()
```

Comment:
__init__ should be called only once on object creation, it is not recommended to call it multiple times within the class